### PR TITLE
Add September monthly update

### PR DESCRIPTION
**Pull request by kagamiori**
Differential Revision: D49847351

### CODE REVIEW

File: velox/docs/monthly-updates/september-2023.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for UNKNOWN types to VectorHasher::hash :pr:`6435`
+* Add support for writing sorted buckted hive table :pr:`6142`
+* Extend Type with isOrderable() and isComparable() :pr:`6770`
+
+
+Presto Functions
+================
+
+* Add bitwise_xor_agg aggregate function :pr:`6705`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:32]: - Let's group scalar, aggregate and window functions together. Let's also group casts together.
- Let's add links to function names using :func:

File: velox/docs/monthly-updates/september-2023.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize try_cast performance by reducing exception throws :pr:`5913`
+* Reduce peak memory usage by roughly half in HiveDataSource :pr:`6601`
+* Tune memory arbitration and spilling for Velox batch user case :pr:`6789`
+
+
+Build Systems
+=============
+
+* Install dependencies needed to support Azure Storage ABFS Connector :pr:`6418`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:86]: Can we list these dependencies explicitly? CC: @kgpai

File: velox/docs/monthly-updates/september-2023.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+* Add comparison Spark functions :pr:`5569`
+* Add support for all patterns in CAST VARCHAR to DATE :pr:`5844`
+* Add support for :spark:func:`rand` function with seed specified :pr:`6112`
+
+
+Hive Connector
+==============
+
+* Add native Parquet writer :pr:`6586`
+* Add S3FileSystem Write support :pr:`6021`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:59]: perhaps, Add support for writing query results to S3

File: velox/docs/monthly-updates/september-2023.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+* Add :doc:`Chapter 2 of programming guide </programming-guide/chapter02>`
+* Add blog post about `learning from optimizing try_cast <https://velox-lib.io/blog/optimize-try_cast.>`_
+
+
+Core Library
+============
+
+* Add BaseVector::containsNullAt API :pr:`6515`
+* Add ContainerRowSerde::compareWithNulls API :pr:`6419`
+* Add default memory reclaimer for exchange client :pr:`6759`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:18]: Let's remove this. Too much of implementation details.

File: velox/docs/monthly-updates/september-2023.rst [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+* Add Operator::initialize API to handle customized operation init :pr:`6422`
+* Add Pacific/Kanton, Europe/Kyiv, America/Ciudad_Juarez time zones :pr:`6670`
+* Add support for Array/Map/RowVector::copyRanges from UNKNOWN source :pr:`6607`
+* Add support for HUGEINT in function signatures :pr:`6799`
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for UNKNOWN types to VectorHasher::hash :pr:`6435`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:24]: Let's remove. Too much details.

File: velox/docs/monthly-updates/september-2023.rst [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+* Add support for writing sorted buckted hive table :pr:`6142`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:25]: typo: buckted -> bucketed

File: velox/docs/monthly-updates/september-2023.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+* Extend Type with isOrderable() and isComparable() :pr:`6770`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:26]: nit: Add Type::isOrderable() and Type::isComparable() APIs

**Review by mbasmanova — COMMENTED** [review]
@kagamiori Wei, thank you for putting together this update. Some comments below.

File: velox/docs/monthly-updates/september-2023.rst (hunk 2 above)
Comment (kagamiori) [reply velox/docs/monthly-updates/september-2023.rst:83]: This PR installs azure-sdk-for-cpp and its dependencies openssl and perl-IPC-Cmd.

**Comment by kagamiori** [comment]
> @kagamiori Wei, thank you for putting together this update. Some comments below.

Hi @mbasmanova, Thank you for reviewing. I've addressed your comments and updated the PR. Please take a look. Thanks!

File: velox/docs/monthly-updates/september-2023.rst (hunk 2 above)
Comment (kgpai) [reply velox/docs/monthly-updates/september-2023.rst]: Thanks , azure/azure-sdk-for-cpp azure-storage-blobs_12.8.0 is the main dependency. We can say azure storage sdk.

**Review by kgpai — COMMENTED** [review]
Thanks Wei, looks great.

File: velox/docs/monthly-updates/september-2023.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+* Add comparison Spark functions :pr:`5569`
+* Add support for :spark:func:`rand` function with seed specified :pr:`6112`
+* Add support for all patterns in CAST VARCHAR to DATE :pr:`5844`
+
+
+Hive Connector
+==============
+
+* Add support for writing query results to S3 :pr:`6021`
+* Fix Parquet requestType to align with scan table schema :pr:`6404`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:57]: It is not clear what this means. Would it be possible to re-write in user-friendly terms?

File: velox/docs/monthly-updates/september-2023.rst [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+
+Performance and Correctness
+===========================
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:64]: Should this go into Presto Functions section?

File: velox/docs/monthly-updates/september-2023.rst [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
+* Fix handling null string in :func:`json_extract` to follow Presto's behavior :pr:`6439`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:66]: Move to Presto Functions section?

File: velox/docs/monthly-updates/september-2023.rst [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+* Fix :func:`min`/:func:`max`/:func:`min_by`/:func:`max_by`` Presto aggregates when input contains nested nulls :pr:`6723`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:67]: +1 

Move to Presto Functions section?

File: velox/docs/monthly-updates/september-2023.rst [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+* Fix :func:`set_union` Presto aggregate to follow Presto's behavior :pr:`6800`
+* Fix tracking of the number of processed rows in Window operator :pr:`6584`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:69]: Remove. Too much detail.

File: velox/docs/monthly-updates/september-2023.rst [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+* Fix :func:`url_extract_path` Presto function :pr:`6657`
+* Optimize exception handling in SIMDJsonExtractor parse and extract :pr:`6776`
+* Optimize large chunk memory allocation in stream arena :pr:`6547`
+* Optimize :func:`reduce_agg` aggregation :pr:`6536`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:73]: Remove.

File: velox/docs/monthly-updates/september-2023.rst [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize TRY_CAST performance by reducing exception throws :pr:`5913`
+* Reduce peak memory usage by roughly half in HiveDataSource :pr:`6601`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:76]: Move to Hive Connector section

File: velox/docs/monthly-updates/september-2023.rst (hunk 2 above)
Comment (kagamiori) [reply velox/docs/monthly-updates/september-2023.rst]: It also build sdk/core/azure-core though.

File: velox/docs/monthly-updates/september-2023.rst [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+============
+
+* Add BaseVector::containsNullAt API :pr:`6515`
+* Add ContainerRowSerde::compareWithNulls API :pr:`6419`
+* Add Operator::initialize API to handle customized operation init :pr:`6422`
+* Add Pacific/Kanton, Europe/Kyiv, America/Ciudad_Juarez time zones :pr:`6670`
+* Add support for Array/Map/RowVector::copyRanges from UNKNOWN source :pr:`6607`
+* Add support for HUGEINT in function signatures :pr:`6799`
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for writing sorted bucketed hive table :pr:`6142`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:23]: Can you move this to Hive Connector section?

File: velox/docs/monthly-updates/september-2023.rst [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+* Add support for UNKNOWN in :func:`map_from_entries` function :pr:`6353`
+* Add support for VARBINARY input to :func:`length` function :pr:`6795`
+* Add CAST DECIMAL to VARCHAR :pr:`6210`
+* Add CAST TIMESTAMP WITH TIME ZONE to and from TIMESTAMP :pr:`6529`
+* Add :func:`bitwise_xor_agg` aggregate function :pr:`6705`
+* Add :func:`geometric_mean` aggregate function :pr:`6678`
+* Add :func:`reduce_agg` aggregate function :pr:`6482`
+* Add support for UNKNOWN type in :func:`arbitrary` aggregate function :pr:`6557`
+* Add support for complex types to compare argument of :func:`min_by` and :func:`max_by` :pr:`6605`
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:42]: "to follow Presto's behavior" is redundant. Please, remove here and in other places.

File: velox/docs/monthly-updates/september-2023.rst [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+* Add support for writing query results to S3 :pr:`6021`
+* Pass TableScan output type to Parquet column reader :pr:`6404`
+* Reduce peak memory usage by roughly half in HiveDataSource :pr:`6601`
+
+
+Performance and Correctness
+===========================
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:69]: Move this to Hive Connector

File: velox/docs/monthly-updates/september-2023.rst [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:70]: "CPU stats tracking" -> "CPU time reporting"

File: velox/docs/monthly-updates/september-2023.rst (hunk 8 above)
Comment (kagamiori) [reply velox/docs/monthly-updates/september-2023.rst]: According to my understanding, this PR passes the expected output type of TableScan to Parquet column reader (https://github.com/facebookincubator/velox/issues/5770). I can double check with @Yuhta tomorrow.

File: velox/docs/monthly-updates/september-2023.rst [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+* Optimize exception handling in SIMDJsonExtractor parse and extract :pr:`6776`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:71]: Perhaps, replace " SIMDJsonExtractor parse and extract " with the names of Presto functions that use this code path. Consider moving to Presto Functions section.

File: velox/docs/monthly-updates/september-2023.rst [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+* Optimize large chunk memory allocation in stream arena :pr:`6547`
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize TRY_CAST performance by reducing exception throws :pr:`5913`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:74]: typos in "reducing exception throws "

File: velox/docs/monthly-updates/september-2023.rst [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+* Tune memory arbitration and spilling for Velox batch use cases :pr:`6789`
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/september-2023.rst:75]: Group all spilling updates together.

**Review by mbasmanova — COMMENTED** [review]
Looks great % a few remaining nits.

File: velox/docs/monthly-updates/september-2023.rst (hunk 19 above)
Comment (kagamiori) [reply velox/docs/monthly-updates/september-2023.rst]: All SIMD JSON functions would be affected and all our JSON functions except json_format and json_parse are SIMD JSON functions. I updated it to "Optimize exception handling in SIMD JSON functions".

**Review by mbasmanova — APPROVED** [review]
Thank you, Wei.

File: velox/docs/monthly-updates/september-2023.rst [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+* Add :func:`geometric_mean` aggregate function :pr:`6678`
+* Add :func:`reduce_agg` aggregate function :pr:`6482`
+* Add support for UNKNOWN type in :func:`arbitrary` aggregate function :pr:`6557`
+* Add support for complex types to compare argument of :func:`min_by` and :func:`max_by` :pr:`6605`
+* Fix :func:`array_remove` null handling :pr:`6424`
+* Fix handling null string in :func:`json_extract` :pr:`6439`
+* Fix :func:`url_extract_path` function :pr:`6657`
+* Fix :func:`min`, :func:`max`, :func:`min_by`, and :func:`max_by` aggregates when input contains nested nulls :pr:`6723`
+* Fix :func:`set_union` aggregate :pr:`6800`
+* Fix :func:`lead` and :func:`lag` window function for int64 offset :pr:`6463`
```
Comment (aditi-pandit) [inline velox/docs/monthly-updates/september-2023.rst:46]: This update isn't needed really.

**Review by aditi-pandit — APPROVED** [review]
Thanks @kagamiori for the updates !

**Review by xiaoxmeng — COMMENTED** [review]
@kagamiori LGTM. Thanks!

**Pull request closed without merging**